src/sentry/features/handler.py

```diff
@@ -14,3 +14,29 @@ def __call__(self, feature, actor):
 
     def has(self, feature, actor):
         raise NotImplementedError
+
+    def has_for_batch(self, batch):
+        # If not overridden, iterate over objects in the batch individually.
+        return {
+            obj: self.has(feature, batch.actor)
+            for (obj, feature) in batch.get_feature_objects().items()
+        }
+
+
+# It is generally better to extend BatchFeatureHandler if it is possible to do
+# the check with no more than the feature name, organization, and actor. If it
+# needs to unpack the Feature object and examine the flagged entity, extend
+# FeatureHandler directly.
```

Comment:
Will there actually be any `FeatureHandlers` that aren't `BatchFeatureHandler` once your getsentry pr is merged? Wondering whether we should just remove the old base class and assume that everything should be batched going forward.

Reply:
There will be, but not a lot.

I didn't count before, but there are actually only two handlers that still depend on the `Feature` object, namely in order to call:
* `feature.project.flags` (in `IssueAlertsTargetingHandler`)
* `feature.plugin.get_plugin_type()` (currently in `SegmentedProjectFeatureHandler` and broken out into a new handler in https://github.com/getsentry/getsentry/pull/3915)

Think those are fixable/expendable?

Reply:
I don't totally understand how `SegmentedProjectFeatureHandler` is used, or how a plugin even gets passed to that type of feature. I'd guess that we might be able to remove this, but we'd have to spend some time investigating to be sure.

`IssueAlertsTargetingHandler` can definitely be removed. It was just in place for a limited time transition. I can remove that in the next couple days.